SLADIM-131

JonasJW opened the pull request:
## Preparations
Please check if the PR fulfills these requirements
- [x] Readme (How to Install, Run, Use and Configure) wurde aktualisiert falls nötig
- [x] Das Feature ist dokumentiert
- [x] Alle nicht getter und setter Methoden sind dokumentiert
- [ ] Es wurden Unit tests für neue Features erstellt
- [ ] Alle Tests müssen erfolgreich sein
- [x] Mindestens ein Reviewer muss den Pull Request akzeptieren

## :ticket: Description


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Update of cpu-monitor-models
- Checking for minimum cpu value


* **What is the current behavior?** (You can also link to an open issue here)
- old cpu-monitor-models
- Only checking for critical cpu threshold


* **What is the new behavior (if this is a feature change)?**



* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)



* **Other information**:


## :lock: Closes
Which issue(s) is (are) being closed by the PR?

JonasJW committed:
SLADIM-131

JonasJW committed:
doc

JonasJW committed:
conflict

JonasJW committed:
added test

JonasJW committed:
minimum cpu functionality and cpu models version update

JonasJW committed:
test update

KaiBraun committed:
bug (half)-fix

stiesssh committed:
added 'todo' to e2e test

stiesssh reviewed (changes requested):
(1) wie vorhin bereits angesprochen: entfernt den initial endpoint
(2) einen sinnvollen test implementieren. da bei diesem PR u.a. eine prüfung des minimalen cpu wertes eingeführt wurde, wäre es z.b. sinnvoll, zu testen, dass beim unterschreiten des minimumalen wertes tatsächlich das dazu passende log erstellt wird. in wie fern das möglich ist, habe ich mir noch nicht angeschaut. evtl. revidiere ich diesen vorschlag später.

buehlefs commented:
Da diese PR eigentlich noch zum letzten Sprint zählt sollten wir die fehlende Testabdeckung von dem Feature lieber als Issue für den nächsten Sprint mit aufnehemen.

stiesssh commented:
oke. was ist mit dem initialen endpoint? darf der auch drinnen bleiben?

buehlefs commented:
Der muss wie besprochen noch raus bevor die PR gemerged werden kann.

stiesssh commented:
oke :)

KaiBraun committed:
temporarily commented out initialEndpoint

Temporaliy commented out so test passes. InitialEndpoint can later be added via the config as soon as config exists for this service

stiesssh commented:
nicht nur auskommentieren, komplett entfernen! (so war zumindest mein verständnis der sache?)
das ganze auskommentieren sorgt nur für eine unmenge an auskommentierten code, der die lesbarkeit verschlechtert.

desweiteren war ich der ansicht das auch die zeilen 9 bis 15, also die definition dieses 'initialEndpoint' entfernt werden sollten? 
es erschließt sich mit schlicht nicht, warum dieser initail endpoint existiert. wenn ihr ihn also im code behalten wollt, möchte ich eine stichhaltige begründung seiner existenz.

KaiBraun committed:
Added tests for monitorService

KaiBraun committed:
Remove initial endpoint

Tried moving initial endpoint to config but didnt work

JonasJW committed:
Merge branch 'Dev' into SLADIM-131
